packages/apps-engine/src/definition/messages/ISystemMessage.ts

```diff
@@ -0,0 +1,79 @@
+import type { IMessage } from '.';
+
+export type SystemMessageType =
+    | 'voip-call-started'
+    | 'voip-call-declined'
+    | 'voip-call-on-hold'
+    | 'voip-call-unhold'
+    | 'voip-call-ended'
+    | 'voip-call-duration'
+    | 'voip-call-wrapup'
+    | 'voip-call-ended-unexpectedly'
+    | 'removed-user-from-team'
+    | 'added-user-to-team'
+    | 'ult'
+    | 'user-converted-to-team'
+    | 'user-converted-to-channel'
+    | 'user-removed-room-from-team'
+    | 'user-deleted-room-from-team'
+    | 'user-added-room-to-team'
+    | 'ujt'
+    | 'livechat_navigation_history'
+    | 'livechat_transfer_history'
+    | 'livechat_transcript_history'
+    | 'livechat_video_call'
+    | 'livechat_transfer_history_fallback'
+    | 'livechat-close'
+    | 'livechat_webrtc_video_call'
+    | 'livechat-started'
+    | 'omnichannel_priority_change_history'
+    | 'omnichannel_sla_change_history'
+    | 'omnichannel_placed_chat_on_hold'
+    | 'omnichannel_on_hold_chat_resumed'
+    | 'otr'
+    | 'otr-ack'
+    | 'user_joined_otr'
+    | 'user_requested_otr_key_refresh'
+    | 'user_key_refreshed_successfully'
+    | 'e2e'
+    | 'uj'
+    | 'ul'
+    | 'ru'
+    | 'au'
+    | 'mute_unmute'
+    | 'r'
+    | 'ut'
+    | 'wm'
+    | 'rm'
+    | 'subscription-role-added'
+    | 'subscription-role-removed'
+    | 'room-archived'
+    | 'room-unarchived'
+    | 'room_changed_privacy'
+    | 'room_changed_description'
+    | 'room_changed_announcement'
+    | 'room_changed_avatar'
+    | 'room_changed_topic'
+    | 'room_e2e_enabled'
+    | 'room_e2e_disabled'
+    | 'user-muted'
+    | 'user-unmuted'
+    | 'room-removed-read-only'
+    | 'room-set-read-only'
+    | 'room-allowed-reacting'
+    | 'room-disallowed-reacting'
+    | 'command'
+    | 'videoconf'
+    | 'message_pinned'
+    | 'message_pinned_e2e'
+    | 'new-moderator'
+    | 'moderator-removed'
+    | 'new-owner'
+    | 'owner-removed'
+    | 'new-leader'
+    | 'leader-removed'
+    | 'discussion-created';
+
+export interface ISystemMessage extends IMessage {
```

Comment:
Should we go with a different type altogether? Given the difference is only one field, I can't say I see the benefits of creating a new one instead of adding it to the existing `IMessage`. Curious as to what was your reasoning here :)

Reply:
Yea, I was thinking about having the system message be a separate thing from IMessage because I thought the data structure would be really different, but in the end, the only difference is the type field, I'll change this